lib/puppet/application/preview.rb

```diff
@@ -421,6 +422,32 @@ def clean
     @exit_code = 0
   end
 
+  def init_node_names_for_compile
+    given_names = options[:nodes]
+    raise UsageError, 'No node(s) given to perform preview compilation for' if given_names.nil? || given_names.empty?
+
+    # If the face action 'node status' exists (provided by PuppetDB), then purge inactive and non existent nodes from the list
+    node_face = Puppet::Interface[:node, :current]
+    if node_face.respond_to?(:status)
```

Comment:
Think test should include if PDB is in actual use. This test will make calls even if PDB is just installed, even if not configured to be used.
It needs to include a test if PDB is used for storeconfigs (as is already done in catalog preview for the purpose of using special version of PDB terminus).